src/resultAsync.ts

```diff
@@ -99,34 +105,53 @@ export class ResultAsync<TValue = Unit, TError = string> {
    */
   static try<TError = string>(
     promise: Promise<void>,
-    errorHandler: FunctionOfTtoK<unknown, Some<TError>>
+    errorHandler:
+      | FunctionOfTtoK<unknown, Some<TError>>
+      | AsyncFunctionOfTtoK<unknown, Some<TError>>
   ): ResultAsync<Unit, TError>;
   static try<TValue = Unit, TError = string>(
     promiseOrFunction:
       | Promise<Some<TValue>>
       | Promise<void>
       | AsyncFunctionOfT<Some<TValue>>
       | AsyncAction,
-    errorHandler: FunctionOfTtoK<unknown, Some<TError>>
+    errorHandler:
+      | FunctionOfTtoK<unknown, Some<TError>>
+      | AsyncFunctionOfTtoK<unknown, Some<TError>>
   ): ResultAsync<TValue, TError> {
     if (isPromise(promiseOrFunction)) {
       return new ResultAsync(
         promiseOrFunction
           .then((value) => Result.success<TValue, TError>(value!))
-          .catch((error) => Result.failure(errorHandler(error)))
+          .catch(unwrapHandledError(errorHandler))
       );
     }
 
     try {
       return new ResultAsync(
         promiseOrFunction()
           .then((value) => Result.success<TValue, TError>(value!))
-          .catch((error) => Result.failure(errorHandler(error)))
+          .catch(unwrapHandledError(errorHandler))
       );
     } catch (error: unknown) {
-      return new ResultAsync(
-        Promise.resolve(Result.failure(errorHandler(error)))
-      );
+      return new ResultAsync(unwrapHandledError(errorHandler)(error));
+    }
+
+    function unwrapHandledError(
```

Comment:
There might be a benefit to making this a function private to the module but not something that closes over the state of `try`.

Reply:
Moved this to the bottom of the module and re-used it in other methods 🍻